internal/controller/promotions/argocd.go

```diff
@@ -145,8 +146,15 @@ func (r *reconciler) authorizeArgoCDAppUpdate(
 			app.Namespace,
 		)
 	}
-	allowedNamespace, allowedName := tokens[0], tokens[1]
-	if stageMeta.Namespace != allowedNamespace || stageMeta.Name != allowedName {
+	allowedNamespaceGlob, err := glob.Compile(tokens[0])
+	if err != nil {
+		return permErr
```

Comment:
Nit: While it's safe to say mutation of the App isn't authorized in this case, the underlying error here is a failure to compile the glob expression and it's being completely suppressed.

imho, we should either wrap and return the glob compilation error, or if we still want to return `permErr`, then at least log the compilation error.

Same for the name glob.

Reply:
I agree